node/core/runtime-api/src/lib.rs

```diff
@@ -341,6 +322,13 @@ where
 	Client::Api: ParachainHost<Block> + BabeApi<Block> + AuthorityDiscoveryApi<Block>,
 {
 	loop {
+		// This adds some back pressure when the subsystem is running at `MAX_PARALLEL_REQUESTS`.
+		if subsystem.is_busy() {
+			// Since we are not using any internal waiting queues, we need to wait for exactly
+			// one request to complete before we can read the next one from the overseer channel.
+			let _ = subsystem.poll_requests().await;
+		}
```

Comment:
nit: can we move this into an `async wait_for_free_worker()`

Reply:
and shouldn't we still be processing `Signal`s while waiting?

Reply:
Note that this is at risk of the odd case between `is_busy` and all queries finishing on time to run into the `if` case in `poll_requests` and be stuck there, recovering on the next `wake`, which should occur for the next message to be sent. Happy to elaborate.

Reply:
Added a comment to clear this up.